### PR TITLE
Tests for display-name option

### PR DESCRIPTION
**Pull request by jstavel**
There are two tests in the PR:
-  test_register_with_display_name
- test_register_twice_with_different_display_name

I've added a function 'fetch-from-inventory' to download inventory records from insights service.
I've started using subtests pytest plugin - see the second test in a list above.
subtests help me to separate logs for each part of a test. It is easier to troubleshoot a test - log messages are separated by 'subtest' message - at least.
another reason for using subtests is that a code is splitted into parts visually too.

### CODE REVIEW

**Comment by jstavel** [comment]
there is a code block that varies depending on environment - a message from register command varies for 'stage' or 'prod' environment.

File: integration-tests/test_display_name_option.py
Comment (ptoscano) [inline integration-tests/test_display_name_option.py:22]: this is already asserted by the basic registration tests, I don't think there is the need to repeat the checks here too

File: integration-tests/test_display_name_option.py
Comment (ptoscano) [inline integration-tests/test_display_name_option.py:52]: ditto

**Review by ptoscano — CHANGES_REQUESTED** [review]
I'm not sure what this "second registration" actually refers to: the existing `test_double_registration` already asserts that trying to register on an already registered system does nothing. This step here in practice is simply `insights-client --display-name=...`, so I'd do that only.

Not doing a "double registration" simplifies things a bit, and also removes the need to use the subtest plugin.

-----

Regarding the `fetch_from_inventory` fixture: while I see where this is coming, I think we need a more generic way to query the Insights inventory, rather than this ad-hoc fixture. I have a draft of work for this to be added in pytest-client-tools, so please hold this off for now.

-----

Thanks!

**Comment by jstavel** [comment]
added another tests:
- test_display_name .. to verify that an application can set display-name
- test_invalid_display_name

File: integration-tests/test_display_name_option.py
Comment (jstavel) [reply integration-tests/test_display_name_option.py]: this is a case when user register a system. He tries to register the system once more, with different display-name. Even this case may not make sense - but it may happen. So I think it is worth automating it.

Comment (Lorquas) [reply integration-tests/test_display_name_option.py]: Lets do this: 

1.  Create a jira to create `fetch_from_inventory`  functionality in client-tools, and block this on that card.
2.  remove it from this PR
3. merge this once thats working.

I think the rest of this is fine as of the latest change.

**Pull request merged**
